lib/ohai/plugins/linux/virtualization.rb

```diff
@@ -188,6 +216,18 @@ def nova_exists?
         # But, this could overwrite virtualization[:systems][:lxc] = "guest"
         # If so, we may need to look further for a differentiator (OHAI-573)
         virtualization[:systems][:lxc] = "host"
+      elsif init_environ_content =~ %r{\x00?container=lxc\x00?}
```

Comment:
Shouldn't that be: `%r{(^|\x00)container=lxc(\x00|$)}`? Otherwise the nulls aren't really doing anything useful.